src/challenge/signals.py

```diff
@@ -9,3 +9,9 @@
 def challenge_save(sender, instance, **kwargs):
     new_index = caches["default"].get("challenge_mod_index", 0) + 1
     caches["default"].set("challenge_mod_index", new_index)
+
+
+@receiver(post_delete, sender=Challenge)
```

Comment:
Can't we attach this `@receiver` to the invalidating function above, too?

Also, judging from [the docs](https://docs.djangoproject.com/en/3.2/topics/signals/#django.dispatch.receiver), you can also pass a list of signals.